modules/consensus/synchronize.go

```diff
@@ -246,16 +257,106 @@ func (cs *ConsensusSet) relayBlock(conn modules.PeerConn) error {
 		return err
 	}
 
-	// Submit the block to the consensus set.
+	// Submit the block to the consensus set and broadcast it.
 	err = cs.AcceptBlock(b)
 	if err == errOrphan {
 		// If the block is an orphan, try to find the parents. The block
 		// received from the peer is discarded and will be downloaded again if
 		// the parent is found.
+		// TODO: log error returned if non-nill?
+		go cs.gateway.RPC(modules.NetAddress(conn.RemoteAddr().String()), "SendBlocks", cs.threadedReceiveBlocks)
+	}
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// relayHeader is an RPC that accepts a block header from a peer.
+func (cs *ConsensusSet) relayHeader(conn modules.PeerConn) error {
+	// Decode the block header from the connection.
+	var h types.BlockHeader
+	err := encoding.ReadObject(conn, &h, types.BlockHeaderSize)
+	if err != nil {
+		return err
+	}
+
+	// Submit the header to the consensus set.
+	err = cs.managedAcceptHeader(h)
+	switch err {
+	case errOrphan:
+		// If the header is an orphan, try to find the parents.
+		// TODO: log error returned if non-nill?
 		go cs.gateway.RPC(modules.NetAddress(conn.RemoteAddr().String()), "SendBlocks", cs.threadedReceiveBlocks)
+	case errFutureTimestamp:
+		// Ignore errFutureTimestamp so that the block is downloaded and passed to
+		// managedAcceptHeader, which will spawn a go thread that waits until the
+		// timestamp would be valid before trying to accept it again.
+		fallthrough
```

Comment:
since we're removing errFutureTimestamp from the set of errors that acceptHeader can return, there's no need for this case or fallthrough.